test/runtests.jl

```diff
@@ -2,8 +2,8 @@ using AwkwardArray
 using JSON
 using Test
 
-@testset "AwkwardArray.jl" begin
     ### PrimitiveArray #######################################################
+@testset "PrimitiveArray" begin
```

Comment:
On the use of multiple `@testset`, it was simply because I didn't know and didn't look up a reference. The repo was set up with Julia's template and I just extrapolated from that.

Similarly, I think that the source code should be in separate files, but I didn't look into how that works (how dependencies between Julia source files are managed). I still plan to do that—I just had to get this initial prototype done as quickly as possible before cycling around to other to-do items.

Reply:
usually you just split them and have a bunch of `include()` at the top level file, in the order you want